spec.emu

```diff
@@ -309,8 +310,14 @@ contributors: Ujjwal Sharma, Younies Mahmoud
               1. Let _nf_ be ! Construct(%NumberFormat%, &laquo; _durationFormat_.[[Locale]], _nfOpts_ &raquo;).
               1. Let _dataLocale_ be _durationFormat_.[[DataLocale]].
               1. Let _dataLocaleData_ be %DurationFormat%.[[LocaleData]].[[&lt;_dataLocale_&gt;]].
-              1. Let _num_ be ! FormatNumeric(_nf_, 𝔽(_value_)).
-              1. Append the new Record { [[Type]]: _unit_, [[Value]]: _num_} to the end of _result_.
+              1. If _separated_ is *false*, then
+                1. Append a new empty List to the end of _result_.
```

Comment:
Question: Why do we need this intermediate structure? Why not append the number parts directly to `_result_`?

Reply:
@sffc it's needed to solve the bug pointed out by @FrankYFTang [here](https://github.com/tc39/proposal-intl-duration-format/issues/114). Basically, we need to keep track of every "element" of the list manually and break them into sub-lists into the main `_result_` list. Then when we pass it into `ListFormat`, it'll work as expected and we flatten the result at the end.

If we don't do this sub-list segmentation, every individual numeric part as well as the separator will be treated by `ListFormat` as an individual element of the list as Frank points out in the issue, leading to an unexpected final output.

Reply:
I see.

Note that `CreatePartsFromList` says

> The CreatePartsFromList abstract operation is called with arguments listFormat (which must be an object initialized as a ListFormat) and list **(which must be a [List](https://tc39.es/ecma262/#sec-list-and-record-specification-type) of String values)**

so we may need to edit the ListFormat spec a bit to allow it to take a list of parts as input.